src/systems/systems/src/pressurization/pressure_valve.rs

```diff
@@ -1,44 +1,99 @@
-use crate::simulation::UpdateContext;
+use crate::{shared::ControllerSignal, simulation::UpdateContext};
 
-use super::PressureValveActuator;
+use super::{CabinPressure, OutflowValveActuator, PressurizationOverheadPanel};
 
 use std::time::Duration;
 use uom::si::{f64::*, ratio::percent};
 
+pub(super) enum PressureValveSignal {
+    Open,
+    Close,
+    Neutral,
+}
+
 pub(super) struct PressureValve {
     open_amount: Ratio,
     target_open: Ratio,
     full_travel_time: Duration,
+    manual_travel_time: Duration,
 }
 
 impl PressureValve {
-    pub fn new() -> Self {
+    pub fn new_outflow_valve() -> Self {
         Self {
             open_amount: Ratio::new::<percent>(100.),
             target_open: Ratio::new::<percent>(100.),
             full_travel_time: Duration::from_secs(4),
+            manual_travel_time: Duration::from_secs(60),
         }
     }
 
-    pub fn update<T: PressureValveActuator>(&mut self, context: &UpdateContext, actuator: &T) {
-        self.target_open = actuator.target_valve_position();
-        if self.open_amount < self.target_open {
-            self.open_amount += Ratio::new::<percent>(
-                self.get_valve_change_for_delta(context)
-                    .min(self.target_open.get::<percent>() - self.open_amount.get::<percent>()),
-            );
-        } else if self.open_amount > self.target_open {
-            self.open_amount -= Ratio::new::<percent>(
-                self.get_valve_change_for_delta(context)
-                    .min(self.open_amount.get::<percent>() - self.target_open.get::<percent>()),
-            );
+    pub fn new_safety_valve() -> Self {
+        Self {
+            open_amount: Ratio::new::<percent>(0.),
+            target_open: Ratio::new::<percent>(0.),
+            full_travel_time: Duration::from_secs(1),
+            manual_travel_time: Duration::from_secs(1),
+        }
+    }
+
+    pub fn update(
+        &mut self,
+        context: &UpdateContext,
+        signal: &impl ControllerSignal<PressureValveSignal>,
+    ) {
+        match signal.signal() {
+            Some(PressureValveSignal::Open) => {
+                if self.open_amount < Ratio::new::<percent>(100.) {
+                    self.open_amount += Ratio::new::<percent>(
+                        self.get_manual_change_for_delta(context)
+                            .min(100. - self.open_amount.get::<percent>()),
+                    );
+                }
+            }
+            Some(PressureValveSignal::Close) => {
+                if self.open_amount > Ratio::new::<percent>(0.) {
+                    self.open_amount -= Ratio::new::<percent>(
+                        self.get_manual_change_for_delta(context)
+                            .min(self.open_amount.get::<percent>()),
+                    );
+                }
+            }
+            Some(PressureValveSignal::Neutral) => (),
+            _ => {
```

Comment:
**suggestion**: By making the `match` "complete" you ensure that whenever you add another enum value, you'll be notified of any places where that value isn't handled yet. Therefore, when dealing with a finite set of values I tend to write out all cases.
```suggestion
            None => {
```